conf/config-default.yaml

```diff
@@ -1,690 +0,0 @@
-#
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-# CAUTION: DO NOT MODIFY DEFAULT CONFIGURATIONS IN THIS FILE.
-# Keep the custom configurations in conf/config.yaml.
-#
-
-apisix:
```

Comment:
This file contains explanations for all configuration fields, some of which cannot even be found in the documentation. These comments and commented-out configurations were previously good examples for modifying `config.yaml`. Now, this part of the information is missing in `config.lua`. 
I am wondering if we can directly use the content from the original `config-default.yaml` as the new `config.yaml`, allowing users to directly modify or remove configurations within it. The `config.lua` will then fill in default values after user updated.

Reply:
I think this is better, but would it be better to name it something like `config.yaml.example` as it is long and not easy to find what is needed.
This way, we will provide a few minimal default configuration files, `config.yaml`, and an example file containing all the configuration documentation, and if the user wants to use the full configuration file, he can replace `config.yaml` with `config.yamlexample`.

Reply:
We don't have documentation for the configuration items in the config file right now, and the jsonschema used for checksums in the code is outdated (out of sync with the config file), which would be another improvement item.
That is, improve the jsonschema and use it to generate human-readable documentation.